### PR TITLE
Add styling to indicate a post is locked

### PR DESCRIPTION
**Pull request by nramadas**
Modified the look of locked posts in a few ways:

**Posts**
* [x] Add a lock icon to the left of the flair or subreddit name, if they exist
* [x] Add a lock icon to the left of the post listing footer, if flair and/or subreddit name *doesn't* exist.

**Comments**
* [x] Remove the ability to comment on a post

:eyeglasses: @ajacksified 

The lock sits next to the flair if it exists:
screen shot 2016-01-11 at 12 16 08 pm

If there isn't any flair, it sits at the bottom:
screen shot 2016-01-11 at 12 17 04 pm

When comments are locked:
screen shot 2016-01-11 at 10 51 23 am

When replies are locked:
screen shot 2016-01-11 at 5 45 06 pm

### CODE REVIEW

File: src/views/components/Comment.jsx [hunk 1]
```diff
@@ -17,6 +17,21 @@ var replyIcon = (
 );
 
 class Comment extends BaseComponent {
+  static propTypes = {
```
Comment (ajacksified) [inline src/views/components/Comment.jsx]: :no_bell: it's not in the styleguide, but we've generally been putting static properties at the bottom of the class definition. Do you have any objections moving these to the bottom?

File: src/views/components/Comment.jsx [hunk 2]
```diff
@@ -243,23 +258,27 @@ class Comment extends BaseComponent {
...
-            <Vote
-              app={ app }
-              setScore={ this.setScore }
-              thing={ comment }
-              token={ token }
-              apiOptions={ apiOptions }
-            />
-          </li>
-          <li>
+          { !locked
```
Comment (ajacksified) [inline src/views/components/Comment.jsx]: We've been avoiding ternaries in favor of assigning react elements to variables, but it _is_ nice to have it inline like this. Thoughts, @schwers / @curioussavage

File: src/views/components/Vote.jsx [hunk 3]
```diff
@@ -110,78 +120,96 @@ class Vote extends BaseComponent {
...
+      <li>
+        <form
+          className='vote-form'
+          action={ '/vote/'+ thing.name }
+          method='post'
+        >
+          <input type='hidden' name='direction' value='1'/>
+          <button
+            type='submit'
+            className={ 'vote text-muted ' + (this.getVoteClass() || '') }
```
Comment (ajacksified) [inline src/views/components/Vote.jsx]: Prefer template strings (here and in `action` above). Also, since getVoteclass returns empty string, you don't need the `||`

File: src/views/components/Vote.jsx [hunk 4]
```diff
@@ -110,78 +120,96 @@ class Vote extends BaseComponent {
...
-  thing: React.PropTypes.oneOfType([
-    propTypes.comment,
-    propTypes.listing,
-  ]).isRequired,
-};
+  render() {
+    return (
+      <ul className='linkbar linkbar-compact'>
+        { !this.props.locked 
+          ? this.renderUpvote()
```
Comment (ajacksified) [inline src/views/components/Vote.jsx]: :no_bell: what do you think about moving the `locked` check to `renderUpvote`, to simplify the `render` function?

File: src/views/pages/listing.jsx [hunk 5]
```diff
@@ -254,15 +263,17 @@ class ListingPage extends BasePage {
...
-            apiOptions={ apiOptions }
-            thingId={ listing.name }
-            user={ user }
-            token={ token }
-            app={ app }
-            ctx={ ctx }
-            onSubmit={ this.onNewComment }
-          />
+          { listing.locked
+            ? <div className="listing-content__locked">Comments are locked</div>
```
Comment (ajacksified) [inline src/views/pages/listing.jsx]: single quotes

**Comment by ajacksified** [comment]
:nail_care:

File: src/views/components/Comment.jsx (hunk 2 above)
Comment (curioussavage) [inline src/views/components/Comment.jsx]: I have no strong preference. I used it like this before. Maybe in limited use its fine?

File: src/views/components/Comment.jsx (hunk 1 above)
Comment (nramadas) [inline src/views/components/Comment.jsx]: no strong objection. i would argue there is something nice about reading the prop types first before reading code that uses said props. it gives a clearer picture of what's to come. i don't feel too strongly about it, so if you guys are more comfortable with static properties at the bottom, i'll move them.

File: src/views/components/Vote.jsx (hunk 4 above)
Comment (nramadas) [inline src/views/components/Vote.jsx]: so, this is getting incredibly nitpicky :), but I think it's confusing for a `render*` method to NOT return html. If the `!locked` check were deciding between two templates, I would agree w/ moving into `renderUpvote`. but since there is potential for for no html to be rendered (and getting a `null` instead), i'd rather keep the check out of the `renderUpvote` method.

**Comment by nramadas** [comment]
:haircut:

File: src/views/components/Comment.jsx (hunk 1 above)
Comment (ajacksified) [inline src/views/components/Comment.jsx]: I'd say that for consistency's sake they should be on the bottom, and we can talk about moving everything up as part of a styleguide addition.

File: src/views/components/Listing.jsx [hunk 6]
```diff
@@ -140,6 +140,9 @@ class Listing extends BaseComponent {
     if (subredditNode || flairNode) {
       var row1Node = (
         <div className='Listing-header-row1'>
+          { listing.locked
+            ? <div className="Listing-lock icon-lock"></div>
```
Comment (ajacksified) [inline src/views/components/Listing.jsx]: single quotes

File: src/views/components/Listing.jsx [hunk 7]
```diff
@@ -210,9 +213,18 @@ class Listing extends BaseComponent {
...
+    const hasFlair =
+      ListingContent.isNSFW(listing) ||
+      linkFlairText ||
+      (!hideSubredditLabel && subreddit);
+
     return (
       <footer className='Listing-footer'>
         <ul className='linkbar text-muted'>
+          { !hasFlair && listing.locked
+            ? <div className="Listing-lock-large listing-footer-icon icon-lock"></div>
```
Comment (ajacksified) [inline src/views/components/Listing.jsx]: single quotes

File: src/views/components/Vote.jsx [hunk 8]
```diff
@@ -6,6 +6,16 @@ import propTypes from '../../propTypes';
 import BaseComponent from '../components/BaseComponent';
 
 class Vote extends BaseComponent {
+  static propTypes = {
+    // apiOptions: React.PropTypes.object,
```
Comment (ajacksified) [inline src/views/components/Vote.jsx]: nuke or uncomment

File: src/views/components/Vote.jsx [hunk 9]
```diff
@@ -110,78 +120,96 @@ class Vote extends BaseComponent {
...
+    return voteClass;
+  }
+
+  renderUpvote() {
+    const { thing } = this.props;
+    return (
+      <li>
+        <form
+          className='vote-form'
+          action={ '/vote/'+ thing.name }
```
Comment (ajacksified) [inline src/views/components/Vote.jsx]: template string

File: src/views/components/Vote.jsx [hunk 10]
```diff
@@ -110,78 +120,96 @@ class Vote extends BaseComponent {
...
+    );
+  }
 
+  renderDownvote() {
+    const { thing } = this.props;
+    return (
+      <li>
+        <form
+          className='vote-form'
+          action={ '/vote/'+ thing.name }
```
Comment (ajacksified) [inline src/views/components/Vote.jsx]: template string

**Comment by ajacksified** [comment]
:nail_care:

File: src/views/pages/listing.jsx [hunk 11]
```diff
@@ -11,6 +11,15 @@ import Loading from '../components/Loading';
 import TopSubnav from '../components/TopSubnav';
 
 class ListingPage extends BasePage {
+  static propTypes = {
```
Comment (schwers) [inline src/views/pages/listing.jsx]: :no_bell: I like this being at the top of the class, but everywhere else seems to keep it at the bottom. Maybe we can make that change everywhere and add a linter rule possibly (thoughts @ajacksified @curioussavage )

Comment (schwers) [inline src/views/pages/listing.jsx]: Totally missed your convo about this.

File: src/views/components/Vote.jsx [hunk 12]
```diff
@@ -110,78 +120,96 @@ class Vote extends BaseComponent {
     return true;
   }
 
-  render() {
-    var voteClass = '';
+  getVoteClass() {
```
Comment (schwers) [inline src/views/components/Vote.jsx]: This could be an actual getter eg `get voteClass()`

**Comment by tdohz** [comment]
FYI I don't think the design is fully locked for this, and some of the requirements listed on the ticket were incorrect. Sorry about that, I wasn't keeping a close eye on it, but in any case there will need to be a few changes - at a minimum, voting should **not** be disabled on locked posts.

File: src/views/components/Comment.jsx (hunk 2 above)
Comment (schwers) [inline src/views/components/Comment.jsx]: I'd prefer avoiding multiline ternaries. I think when one of the cases is null it does make things nice and maybe that's an exception?. Pulling them out into assignments or render functions, to me, makes the overall template more readable and clean. 

One thing that seems like a code smell to me in this case is both of the ternaries checking !locked. I think'd I'd prefer one check for locked and then both li's, but on the other hand you have to then either have to have two assignments or an array (both seem clunky).

Thoughts?

**Comment by nramadas** [comment]
:haircut:

**Comment by tdohz** [comment]
:+1: This looks good to me, especially as a temp solution while we build out the toasts. I don't think Erica has access to this repo so I'll just do a quick check with her in slack.

File: src/views/pages/listing.jsx [hunk 13]
```diff
@@ -254,30 +254,32 @@ class ListingPage extends BasePage {
...
-            user={ user }
-            token={ token }
-            app={ app }
-            ctx={ ctx }
-            onSubmit={ this.onNewComment }
-          />
+          { listing.locked
+            ? <div className='listing-content__locked'>Comments are locked</div>
+            : <CommentBox
+                apiOptions={ apiOptions }
```
Comment (schwers) [inline src/views/pages/listing.jsx]: :no_bell: Linter gives a warning on the spaces here because it doesn't know how to deal with the ternary increasing the indentation. Apparently
 ``` 
{ listing.locked
  ? // (can be here or on the previous line)
  <div className='listing-content__locked'>Comments are locked</div>
  :
  <CommentBox
    apiOptions={ apiOptions }
    thingId={ listing.name }
    user={ user }
    token={ token }
    app={ app }
    ctx={ ctx }
    onSubmit={ this.onNewComment }
  /> }
``` is valid and linter warning free

File: src/views/components/Listing.jsx [hunk 14]
```diff
@@ -140,6 +140,9 @@ class Listing extends BaseComponent {
     if (subredditNode || flairNode) {
       var row1Node = (
         <div className='Listing-header-row1'>
+          { listing.locked
+            ? <div className='Listing-lock icon-lock'></div>
```
Comment (schwers) [inline src/views/components/Listing.jsx]: small nit: can you make this self-closing?

File: src/views/components/Listing.jsx [hunk 15]
```diff
@@ -210,9 +213,18 @@ class Listing extends BaseComponent {
...
+    const hasFlair =
+      ListingContent.isNSFW(listing) ||
+      linkFlairText ||
+      (!hideSubredditLabel && subreddit);
+
     return (
       <footer className='Listing-footer'>
         <ul className='linkbar text-muted'>
+          { !hasFlair && listing.locked
+            ? <div className='Listing-lock-large listing-footer-icon icon-lock'></div>
```
Comment (schwers) [inline src/views/components/Listing.jsx]: same small nit: self-closing preferred

File: src/views/components/Vote.jsx [hunk 16]
```diff
@@ -110,78 +110,100 @@ class Vote extends BaseComponent {
...
+          <input type='hidden' name='direction' value='1'/>
+          <button
+            type='submit'
+            className={ `vote text-muted ${this.voteClass}` }
+            data-vote='up'
+            data-thingid={ thing.name }
+            data-no-route='true'
+            onClick={ this.upvote }
+          >
+            <span className='icon-upvote-circled'></span>
```
Comment (schwers) [inline src/views/components/Vote.jsx]: self-closing nti

File: src/views/components/Vote.jsx [hunk 17]
```diff
@@ -110,78 +110,100 @@ class Vote extends BaseComponent {
...
+          <input type='hidden' name='direction' value='-1'/>
+          <button
+            type='submit'
+            className={ `vote text-muted ${this.voteClass}` }
+            data-vote='down'
+            data-thingid={ thing.name }
+            data-no-route='true'
+            onClick={ this.downvote }
+          >
+            <span className='icon-downvote-circled'></span>
```
Comment (schwers) [inline src/views/components/Vote.jsx]: self-closing nit

**Comment by schwers** [comment]
:nail_care: only style / linter warning nitpicks. otherwise looks good and works from my testing

**Comment by nramadas** [comment]
@schwers :haircut:

File: package.json [hunk 18]
```diff
@@ -9,7 +9,8 @@
     "watch": "gulp watch",
     "dev": "gulp dev",
     "test": "mocha test/index.js",
-    "lint": "eslint src --ext .js,.jsx"
+    "lint": "eslint src --ext .js,.jsx",
+    "fonts": "gulp icon-fonts"
```
Comment (schwers) [inline package.json]: :no_bell: _maybe_ should be in its own patch but just please update the building page in the wiki when this goes out

**Comment by schwers** [comment]
:fish:

File: package.json (hunk 18 above)
Comment (nramadas) [inline package.json]: whoops, i meant to take that out actually
